WordPress/Classes/Services/AccountService.m

```diff
@@ -35,22 +35,20 @@ @implementation AccountService
  */
 - (WPAccount *)defaultWordPressComAccount
 {
-    NSString *uuid = [[NSUserDefaults standardUserDefaults] stringForKey:DefaultDotcomAccountUUIDDefaultsKey];
-    if (uuid.length == 0) {
+    NSURL *myObjectURL = [NSURL URLWithString:[[NSUserDefaults standardUserDefaults] objectForKey:DefaultDotcomAccountIDDefaultsKey]];
+    NSManagedObjectID *myObjectID = [self.managedObjectContext.persistentStoreCoordinator managedObjectIDForURIRepresentation:myObjectURL];
+
+    if (!myObjectID) {
         return nil;
     }
 
-    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Account"];
-    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"uuid == %@", uuid];
-    fetchRequest.predicate = predicate;
-    
     NSError *error = nil;
-    NSArray *fetchedObjects = [self.managedObjectContext executeFetchRequest:fetchRequest error:&error];
+    WPAccount *foundAccount = [self.managedObjectContext existingObjectWithID:myObjectID error:&error];
```

Comment:
I understand that it can change during a migration, but we only need to be not repeated among two instances of WPAccount, that's all, it doesn't matter if it changes. At least I don't see how it would break it.
Care to explain further?